src/core/metrics.cc

```diff
@@ -310,6 +310,46 @@ instance_id_type shard() {
     return sstring("0");
 }
 
+void
+impl::set_metrics_to_replicate(std::vector<replicated_metric_spec> metrics_to_replicate) {
+    for (const auto& spec: metrics_to_replicate) {
+        if (!_value_map.contains(spec.full_name)) {
+            continue;
+        }
+
+        replicate_metric_family(spec);
+    }
+
+    _metrics_to_replicate = std::move(metrics_to_replicate);
+}
+
+void impl::replicate_metric(const replicated_metric_spec& matching_spec,
+                            shared_ptr<registered_metric> metric,
+                            const metric_family& family) const {
+    auto destination_impl = get_local_impl(matching_spec.destination_handle);
```

Comment:
Hold by ref?

Reply:
I'd rather hold by value for a shared_ptr. const& would work, but I don't like the lifetime extension stuff that comes along with it.